kolibri/core/tasks/iceqube/classes.py

```diff
@@ -0,0 +1,172 @@
+            )
+
+        self.func = funcstring
+
+    def get_lambda_to_execute(self):
+        """
+        return a function that executes the function assigned to this job.
+
+        If job.track_progress is None (the default), the returned function accepts no argument
+        and simply needs to be called. If job.track_progress is True, an update_progress function
+        is passed in that can be used by the function to provide feedback progress back to the
+        job scheduling system.
+
+        :return: a function that executes the original function assigned to this job.
+        """
+
+        def y(update_progress_func, cancel_job_func):
+            """
+            Call the function stored in self.func, and passing in update_progress_func
+            or cancel_job_func depending if self.track_progress or self.cancellable is defined,
+            respectively.
+            :param update_progress_func: The callback for when the job updates its progress.
+            :param cancel_job_func: The function that the function has to call occasionally to see
+            if the user wants to cancel the currently running job.
+            :return: Any
+            """
+
+            setattr(current_job_tracker, "job", self)
+
+            func = import_stringified_func(self.func)
+            extrafunckwargs = {}
+
+            args, kwargs = copy.copy(self.args), copy.copy(self.kwargs)
+
+            if self.track_progress:
+                extrafunckwargs["update_progress"] = partial(
+                    update_progress_func, self.job_id
+                )
+
+            if self.cancellable:
+                extrafunckwargs["check_for_cancel"] = partial(
+                    cancel_job_func, self.job_id
+                )
+
+            kwargs.update(extrafunckwargs)
+
+            try:
+                result = func(*args, **kwargs)
+            except Exception as e:
+                # If any error occurs, clear the job tracker and reraise
+                setattr(current_job_tracker, "job", None)
+                raise e
+
+            setattr(current_job_tracker, "job", None)
+
+            return result
+
+        return y
+
+    @property
+    def percentage_progress(self):
+        """
+        Returns a float between 0 and 1, representing the current job's progress in its task.
+        If total_progress is not given or 0, just return self.progress.
+
+        :return: float corresponding to the total percentage progress of the job.
+        """
+
+        if self.total_progress != 0:
+            return float(self.progress) / self.total_progress
+        else:
+            return self.progress
+
+    def __repr__(self):
+        return "<Job id: {id} state: {state} progress: {p}/{total} func: {func}>".format(
+            id=self.job_id,
+            state=self.state,
+            func=self.func,
+            p=self.progress,
+            total=self.total_progress,
+        )
```

Comment:
"classes.py" is a bit non-descriptive for a file name

Reply:
But... it has the classes!

Can address as part of the renaming.